pkg/apis/apis/v1alpha1/types_apiexport.go

```diff
@@ -209,6 +210,40 @@ func (p PermissionClaim) Equal(claim PermissionClaim) bool {
 		p.IdentityHash == p.IdentityHash
 }
 
+// PermissionClaimFrom parses s and returns a PermissionClaim, or an error if parsing was not possible. s is expected
+// to be in the format produced by String(): either <identityHash>:<resource>.<group> or <resource>.<group>.
+func PermissionClaimFrom(s string) (*PermissionClaim, error) {
```

Comment:
this helper surprises me in the middle of the API definition. In which sense is this an API ? Where is it used?

Reply:
It's operating over the literal values in the spec/status, no? That's the sense in which it's part of our API definition?

Reply:
but it is not API. It's a helper used in the controller for whatever string generation (errors, conditions). 

Litmus test for being here in the package: if somebody changes this func, is it worth an API review?

Reply:
it doesn't have to be in here. i'm using the string form inside `sets.String` instances, and then parsing the string back into a PermissionClaim where needed. happy to move this elsewhere or come up with a better approach

Reply:
Hm, I thought since the literal values in the spec & status are the ones being parsed, this *is* an API since those are the only valid formats?

Reply:
I've moved to helpers scoped to the reconciler in my next commit